consensus/src/sync/live/block_queue/queue.rs

```diff
@@ -216,17 +207,32 @@ impl<N: Network> BlockQueue<N> {
                 );
                 return Some(QueuedBlock::TooFarAhead(peer_id));
             }
-        } else if block_number <= macro_height {
+        } else if block_number <= self.current_macro_height {
             // Block is from a previous batch/epoch, discard it.
             log::warn!(
                 "Discarding block {} - we're already at macro block #{}",
                 block,
-                macro_height
+                self.current_macro_height
             );
             block_source.ignore_block(&self.network);
-        } else {
-            // Block is inside the buffer window, put it in the buffer.
+        } else if self.request_component.has_no_pending_requests()
+            || self.current_macro_height == Policy::last_macro_block(block.block_number())
+        {
+            // We only allow a new request missing blocks to start if the block is from the
+            // current batch or if there are no ongoing request.
             self.buffer_and_request_missing_blocks(block, block_source);
+        } else {
+            // If we are on not within the same batch or we already are requesting blocks,
+            // we just buffer it without requesting for blocks.
+            // Any potential gaps will be filled after we sync up to the batch.
+            if self.insert_block_into_buffer(block, block_source.clone()) {
```

Comment:
```suggestion
            if self.insert_block_into_buffer(block, block_source) {
```
Looks like this `clone()` is unnecessary.